File: src/api/CafeRMS.Api/Features/Auth/Permissions.cs
namespace CafeRMS.Api.Features.Auth;

public static class Permissions
{
    public const string OutletManage = "OutletManage";
    public const string TablesManage = "TablesManage";
    public const string ProductsManage = "ProductsManage";
    public const string ModifiersManage = "ModifiersManage";
    public const string MenusManage = "MenusManage";
    public const string TaxRatesManage = "TaxRatesManage";
    public const string PricingManage = "PricingManage";
    public const string SalesChannelsManage = "SalesChannelsManage";
    public const string PromotionsManage = "PromotionsManage";
    public const string EventsManage = "EventsManage";
    public const string UsersManage = "UsersManage";
    public const string RolesManage = "RolesManage";
    public const string PrintoutTemplatesManage = "PrintoutTemplatesManage";
    public const string OrdersView = "OrdersView";
    public const string OrdersManage = "OrdersManage";
    public const string ReportsView = "ReportsView";
    public const string LoyaltyManage = "LoyaltyManage";

    public static readonly string[] All =
    [
        OutletManage, TablesManage, ProductsManage, ModifiersManage,
        MenusManage, TaxRatesManage, PricingManage, SalesChannelsManage,
        PromotionsManage, EventsManage, UsersManage, RolesManage,
        PrintoutTemplatesManage, OrdersView, OrdersManage, ReportsView,
        LoyaltyManage
    ];
}

File: src/web/src/features/auth/permissions.ts
// Mirrors CafeRMS.Api.Features.Auth.Permissions — keep in sync with the C# constants.

export const ALL_PERMISSIONS = [
  "OutletManage",
  "TablesManage",
  "ProductsManage",
  "ModifiersManage",
  "MenusManage",
  "TaxRatesManage",
  "PricingManage",
  "SalesChannelsManage",
  "PromotionsManage",
  "EventsManage",
  "UsersManage",
  "RolesManage",
  "PrintoutTemplatesManage",
  "OrdersView",
  "OrdersManage",
  "ReportsView",
  "LoyaltyManage",
] as const

export type Permission = (typeof ALL_PERMISSIONS)[number]

export const PERMISSION_LABELS: Record<Permission, string> = {
  OutletManage: "Outlet — manage",
  TablesManage: "Tables — manage",
  ProductsManage: "Products — manage",
  ModifiersManage: "Modifiers — manage",
  MenusManage: "Menus — manage",
  TaxRatesManage: "Tax rates — manage",
  PricingManage: "Pricing — manage",
  SalesChannelsManage: "Sales channels — manage",
  PromotionsManage: "Promotions — manage",
  EventsManage: "Events — manage",
  UsersManage: "Users — manage",
  RolesManage: "Roles — manage",
  PrintoutTemplatesManage: "Printout templates — manage",
  OrdersView: "Orders — view",
  OrdersManage: "Orders — manage (close / cancel)",
  ReportsView: "Reports — view",
  LoyaltyManage: "Loyalty — manage",
}
